narwhal/primary/src/proposer.rs

```diff
@@ -241,7 +241,18 @@ impl Proposer {
         self.proposed_headers.insert(this_round, header.clone());
 
         // Update metrics related to latency
-        for (_, _, created_at_timestamp) in digests {
+        for (_digest, _worker_id, created_at_timestamp) in digests {
+            #[cfg(feature = "benchmark")]
+            {
+                // NOTE: This log entry is used to compute performance.
+                tracing::info!(
+                    "Batch {:?} from worker {} took {} seconds to be included in a proposed header",
```

Comment:
1. Let's mention this is batch creation to inclusion in header?
2. I think this should be calculated as `header.created_at - created_at_timestamp`?

Reply:
The two use cases for this timestamp
1. How long does it take for a batch, after creation, to be included in a header?
2. How long does it take for a header to be created?

Will calculate the first as the difference from header creation time to batch creation time. It will come out to the same value because we were calculating the `created_at_timestamp.elapsed_time()` from `now()` at the same time we were generating the header which has its `created_at` time initialized to `now()`

The second will be calculated as the difference of the header `created_at` time and the `created_at_timestamp` of the first batch included in the header.